app/src/main/java/com/limelight/binding/input/driver/UsbDriverService.java

```diff
@@ -248,9 +251,31 @@ else if (kernelVersion.startsWith("4.4.") || kernelVersion.startsWith("4.9.")) {
         }
     }
 
+    public static boolean kernelSupportsXbox360W() {
+        String kernelVersion = System.getProperty("os.version");
+        LimeLog.info("Kernel Version: "+kernelVersion);
+
+        if (kernelVersion == null) {
+            // We'll assume this is some newer version of Android
+            // that doesn't let you read the kernel version this way.
+            return true;
+        }
+        else if (kernelVersion.startsWith("2.") ||
+                kernelVersion.startsWith("3.") ||
+                kernelVersion.startsWith("4.")){
+            // Starting 4.2, Linux kernels support Xbox 360 W but none are AOSP.
+            return false;
```

Comment:
I was at a loss trying to figure out what was going on because https://github.com/torvalds/linux/commit/75b7f05d2798ee3a1cc5bbdd54acd0e318a80396 is clearly present in 4.2+. Looking closer though, this change has a dependency on a kernel build option.

I think the real issue is that these devices have kernels with with `CONFIG_JOYSTICK_XPAD_LEDS` disabled. Without that enabled, the code to set the LED will not be compiled into the kernel. This can happen independently of the kernel version.

So to handle this properly, we can't just use a kernel version check. We have to actually check that the LED devices appear under `/sys/class/leds/xpad[0-9]` to see if LED support is properly enabled.

Reply:
Fixed.